firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -407,6 +407,7 @@ private static void logBadConfigError() {
   private HttpURLConnection openHttpURLConnection(URL url, String apiKey) throws IOException {
     HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection();
     httpURLConnection.setConnectTimeout(NETWORK_TIMEOUT_MILLIS);
+    httpURLConnection.setUseCaches(false);
```

Comment:
Aren't you worried that this might trigger unexpected errors?
Have you tested this already or should we test it maybe?

Reply:
Additionally, are we sure that `POST`s actually respect this value? I would imagine that posts are just never cached

Reply:
Since Android 4.4, it uses okhttp for all Http calls. 

setUseCaches is set to true by default. 

The [ResponseCache](https://developer.android.com/reference/java/net/ResponseCache) implementation decides which resources should be cached, and for how long they should be cached. If a request resource cannot be retrieved from the cache, then the protocol handlers will fetch the resource from its original location. The settings for URLConnection#useCaches controls whether the protocol is allowed to use a cached response.

From the user reported stack trace:
`com.android.okhttp.internal.huc.JavaApiConverter.createOkResponseForCacheGet(JavaApiConverter.java:216)
`

And the [javadocs](https://github.com/square/okhttp/blob/ee5cf502889ed896a67dac57662d19bc469bb876/okhttp-android-support/src/main/java/okhttp3/internal/huc/JavaApiConverter.java#L204-L210) of JavaApiConverter.createOkResponseForCacheGet: **Creates an OkHttp Response using the supplied Request and CacheResponse to supply the data** 

Looks like there is response caching internally. 
 
Regarding POST not being cached: 

From the [rfc](https://www.ietf.org/rfc/rfc2616.txt) : Responses to POST method are not cacheable, unless the response includes appropriate Cache-Control or Expires header fields. 

Cache-Control is set to private for getToken response.

@vkryachko Do you see any issue in disabling the cache?

Reply:
Also, I think we can benefit from setting Cache-Control request header as  **no-cache**